OneCommunityGlobal/HGNRest
# Rahul - Added HGN Form Backend Code

# Description
This PR includes the backend code for the HGN Form application, which allows developers to evaluate their skill sets across various parameters, facilitating collaboration by enabling them to offer or request assistance from others. The backend primarily handles fetching the form questions. In owner mode, the questions can be edited directly from the frontend. This backend implementation includes all the necessary routing and logic to serve and update the questions efficiently.

## Related PR
To test this backend PR you need to checkout the [#2922](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2922) frontend PR.

## Main changes explained
- models/Question.js: Defines the structure of 'Questions'
- routes/questions.js: Contains the necessary routes for loading and editing questions, facilitating frontend-backend communication.
- seed.js: Populates the MongoDB database with initial question data.
- server.js: Establishes the database connection and starts the backend server.

## How to test
1. check into current branch: `git checkout Rahul-HGN-Form-Backend`
2. change directory using: `cd src/hgn-form/backend`
3. update '.env' file. First, create a sample MongoDB database to test this PR. Then add the URL of your database to MONGO_URI in the '.env' file
4. run `npm install` to install the dependencies
5. run `node seed.js` to populate your MongoDB database
6. run `node server.js` to start your backend server
7. Ensure the backend is running correctly and that data is successfully stored in your MongoDB database.
8. Review the code files to verify both the logic and syntax.

## Video Explanation - How to test this PR
https://www.loom.com/share/e8a28ca8345a4938b8348a208311adb1?sid=321f16bb-0e22-48b5-9b3e-18d78589bdfb
